mlflow/utils/autologging_utils.py

```diff
@@ -662,6 +690,26 @@ def call_original(*og_args, **og_kwargs):
     wrap_patch(destination, function_name, safe_patch_function)
 
 
+def _validate_autologging_run(autologging_integration, run_id):
```

Comment:
Could this function not just take a `Run` object instead of its id, since we use the id to immediately get the run object anyway?

Reply:
Good question! There's a bit of a subtle & unfortunate distinction here. The first run object that we fetch the ID from is a fluent active run, which doesn't contain up-to-date tags / metrics / params set during the execution of patch code. The second run object is a Run entity (https://mlflow.org/docs/latest/python_api/mlflow.entities.html#mlflow.entities.Run) fetched via the MLflow client that contains tags (which we need to inspect) / metrics / params.

Reply:
Right, that would make sense. Thanks!